KarhooUISDK/Screens/CheckoutScreen/Checkout/Workers/KarhooCheckoutBookingWorker.swift

```diff
@@ -279,10 +269,21 @@ final class KarhooCheckoutBookingWorker: CheckoutBookingWorker {
 
     // MARK: - Manage payment-related flow
 
-    private func requestNewPaymentMethod(showRetryAlert: Bool = false) {
-        paymentWorker.requestNewPaymentMethod(showRetryAlert: showRetryAlert) { [weak self] result in
-            self?.handleAddNewPaymentMethod(with: result)
+    private func requestNewPaymentMethod(showRetryAlert: Bool) {
+        let baseViewController = UIApplication.shared.topMostViewController() as? BaseViewController
+        if showRetryAlert {
+            baseViewController?.showUpdatePaymentCardAlert(
```

Comment:
I don't like managing view right away from the model layer. 
Would you please add // some sort of comment informing this solution needs to be changes at some point? But not TODO, since it's visible to our DPs.

Reply:
maybe we should just create a ticket for this?